elisp.nix

```diff
@@ -9,15 +9,25 @@ let
   parse = pkgs.callPackage ./parse.nix {};
 in {
   config,
+  alwaysEnsure ? false, # emulate `use-package-always-ensure` behavior
```

Comment:
I think we should default this to `true`

Reply:
I think the old behavior is a bit broken and since most people don't have configs written with it in mind, I think it's okay to break the backwards compatibility.

Reply:
I'm inclined to agree with @talyz that the previous behaviour was incorrect and the new behaviour is more desirable and obvious to newcomers.

Reply:
An addendum to this - the main reason I think it's broken is than if you have packages that _shouldn't_ be ensured, you'll get warnings about missing emacs packages. There may also be other reasons why you don't want packages installed, but nothing I've run into personally; I can imagine it being an issue if you keep a locally modified version of the package in your repo that you want to use instead, though.

Reply:
I think that the behavior should to be the same as for the `use-package-always-ensure` which is `nil` by default. So if I don't use `alwaysEnsure` I have to add `:ensure t` to every package I want Nix to install. Conversely, if I set `alwaysEnsure` to `true` I have to add `:ensure nil` to every package I DON'T want Nix to install.

https://jwiegley.github.io/use-package/keywords/#ensure-pin

Reply:
Just to make this clear: the behavior that @alexvorobiev describes is the currently implemented and merged one: explicitly set `:ensure`s will always override the value of `alwaysEnsure`.